### PR TITLE
Mention build scans in 4.10 release notes 

### PR DESCRIPTION
**Pull request by breskeby**
### Context

Added a section to the release notes mentioning the new defaulit build scan plugin version 1.16.

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -104,6 +104,21 @@ However, we strongly recommend that build and plugin authors use Gradle properti
 Memory usage for up-to-date checking has been improved.
 For the gradle/gradle build, heap usage dropped by 60 MB to 450 MB, that is a 12% reduction.
 
+### Build Scan Plugin default version updated to 1.16
+
+The default build scan plugin version has been updated to. This version provides you new insights of your build, for example:
```
Comment (melix) [inline subprojects/docs/src/docs/release/notes.md]: :x: has been updated
:x: I'd just say "provides you with new insights:"

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -104,6 +104,21 @@ However, we strongly recommend that build and plugin authors use Gradle properti
...
+
+- Sources of slow configuration lifecycle callbacks
```
Comment (melix) [inline subprojects/docs/src/docs/release/notes.md]: :o: This is a bit unclear

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -104,6 +104,21 @@ However, we strongly recommend that build and plugin authors use Gradle properti
...
+- Build script compilation
+- Build deprecation warnings
+- Dependency repositories
+
+in build scans.
```
Comment (melix) [inline subprojects/docs/src/docs/release/notes.md]: Let's drop this line

File: subprojects/docs/src/docs/release/notes.md [hunk 4]
```diff
@@ -104,6 +104,21 @@ However, we strongly recommend that build and plugin authors use Gradle properti
...
+
+You can enable build scans by running your build with `--scan`.
```
Comment (melix) [inline subprojects/docs/src/docs/release/notes.md]: :o: I don't think we need this in release notes.

Comment (breskeby) [reply subprojects/docs/src/docs/release/notes.md]: @melix after consulting with @tomdcc I've reworded the paragraph

Comment (breskeby) [reply subprojects/docs/src/docs/release/notes.md]: @melix and removed the empty promoted feature paragraph

**Pull request merged**